dotcom-rendering/src/lib/scheduler.test.ts

```diff
@@ -0,0 +1,155 @@
+import {
+	schedule,
+	// type ScheduleOptions,
+	setSchedulerConcurrency,
+	// setSchedulerDebugMode,
+	setSchedulerPriorityLastStartTime,
+	// type TaskPriority,
+} from './scheduler';
+
+jest.useFakeTimers();
+
+describe('scheduler', () => {
+	beforeEach(() => {
+		jest.clearAllTimers();
+		jest.clearAllMocks();
+	});
+
+	it('should run scheduled tasks', async () => {
+		const task = jest.fn().mockResolvedValue('task result');
+		const result = await schedule('testTask', task, {
+			priority: 'feature',
+		});
+
+		expect(result).toBe('task result');
+		expect(task).toHaveBeenCalled();
+	});
+
+	it('should not run tasks that should not run', async () => {
+		const taskWillRun = jest.fn().mockResolvedValue('will run result');
+		const taskWillNotRun = jest
+			.fn()
+			.mockResolvedValue('will not run result');
+
+		const willRunResult = schedule('will run', taskWillRun, {
+			priority: 'feature',
+			canRun: () => true,
+		});
+
+		void schedule('will not run', taskWillNotRun, {
+			priority: 'feature',
+			canRun: () => false,
+		});
+
+		expect(taskWillRun).toHaveBeenCalled();
+		expect(taskWillNotRun).not.toHaveBeenCalled();
+
+		jest.runAllTimers();
+
+		expect(await willRunResult).toBe('will run result');
+		expect(taskWillNotRun).not.toHaveBeenCalled();
+	});
+
+	it('should not run tasks after their last start time', () => {
+		setSchedulerPriorityLastStartTime('enhancement', 1000);
+
+		const enhancement = jest.fn().mockResolvedValue('task result');
+		const feature = jest.fn().mockResolvedValue('task result');
+
+		jest.advanceTimersByTime(10000);
```

Comment:
Wondering if we should add an ESLint rule for this.

```suggestion
		jest.advanceTimersByTime(10_000);
```